src/task/TaskManager.ts

```diff
@@ -0,0 +1,71 @@
+import {TaskClient} from "./TaskClient"
+import os from "os"
+import {RunnerOptions, TaskRunner} from "./TaskRunner"
+import {ConductorLogger, DefaultLogger} from "../common/ConductorLogger"
+import {ConductorWorker} from "./Worker"
+
+export interface TaskManagerConfig {
+  logger?: ConductorLogger
+  options?: Partial<RunnerOptions>
+}
+
+const defaultRunnerOptions: Required<RunnerOptions> = {
+  workerID: '',
+  pollInterval: 1000,
+  maxRunner: 1,
+  domain: ''
+}
+
+function workerId (options: Partial<RunnerOptions>) {
+  return options.workerID ?? os.hostname()
+}
+
+export class TaskManager {
+  private tasks: Record<string, Array<TaskRunner>> = {}
+  private readonly client: TaskClient
+  private readonly logger: ConductorLogger
+  private workers: Array<ConductorWorker>
+  private runnerOptions: Required<RunnerOptions>
+
+  constructor(client: TaskClient, workers: Array<ConductorWorker>, config: TaskManagerConfig = {}) {
+    if (!workers) { throw new Error("No workers supplied to TaskManager. Please pass an array of workers.") }
+    this.client = client
+    this.logger = config.logger ?? new DefaultLogger()
+    this.workers = workers
+    const providedOptions = config.options ?? {}
+    this.runnerOptions = {
+      ...defaultRunnerOptions,
+      workerID: workerId(providedOptions),
+    }
+  }
+
+  startPolling = () => {
+    const maxRunner = this.runnerOptions.maxRunner ?? 1
+    this.workers.forEach(worker => {
+      this.tasks[worker.taskDefName] = []
+      for (let i = 0; i < maxRunner; i++) {
+        const runner = new TaskRunner({
+          worker,
+          options: this.runnerOptions,
+          client: this.client,
+          logger: this.logger
+        })
+        // TODO(@ntomlin): right now we aren't handling these promises
+        // which will inevitably lead to chaos
+        runner.startPolling()
```

Comment:
Related to above, I'm not sure if we should be storing and/or catching errors within the promise each `TaskRunner` instance returns to try and build a way to have higher level error handling (e.g. if a critical failure happens, shut all runners down).

Reply:
Update: I think creating an [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) and passing it around would be the cleanest way to implement this. 

We currently have a variation on this with how we check for `isPolling` in taskRunner but this would compose well with `fetch` and goes with the grain in node.

I don't think we have to do that now but it could be a nice enhancement.